qemu/tests/virtio_console.py

```diff
@@ -1792,19 +1765,40 @@ def test_unplugged_restart():
                       method=params.get('virtio_console_method', 'shell'),
                       timeout=720)
         except Exception, details:
-            raise error.TestFail("Fail to reboot VM:\n%s" % details)
+            test.fail("Fail to reboot VM:\n%s" % details)
 
         # TODO: Hotplug ports and verify that they are usable
         # VM is missing ports, which are in params.
         vm.destroy(gracefully=True)
 
-    @error.context_aware
+    @error_context.context_aware
     def test_failed_boot():
         """
         Start VM and check if it failed with the right error message.
         :param cfg: virtio_console_params - Expected error message.
-        """
-        exp_error_message = params['virtio_console_params']
+        :param cfg: qemu_version_pattern - Qemu version pattern for high version.
+        :param cfg: max_ports_invalid - Invalid max ports nums for different versions.
+        :param cfg: max_ports_valid - Valid max ports nums for different versions.
+
+        """
+        max_ports_invalid = params['max_ports_invalid']
```

Comment:
how about put these in one line `max_ports_invalid = params['max_ports_invalid'].split(',')`